### PR TITLE
Add OneSender Webhook notification

### PR DESCRIPTION
**Pull request by hasanbasri1993**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #(issue)

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)
- New feature (non-breaking change which adds functionality)
- Breaking change (a fix or feature that would cause existing functionality to not work as expected)
- Other
- This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

| Event | Before | After |
| ------ | ------|------|
| `UP` | - | image |
| `DOWN` | - | image |
| Certificate-expiry | - | image |
| Testing | - | image |

### CODE REVIEW

File: src/components/notifications/Onesender.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            placeholder="https://xxxxxxxxxxx.com/api/v1/messages"
+            pattern="https?://.+"
+            class="form-control"
+            required
+        />
+    </div>
+
+    <div class="mb-3">
+        <label for="receiver-onesender" class="form-label">{{ $t("TokenOnesender") }}</label>
+        <input
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:17]: Please use `HiddenInput` for secret variables instead ^^

File: src/components/notifications/Onesender.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        <label for="type-receiver-onesender" class="form-label">{{ $t("ReceiverOnesender") }}</label>
+        <input
+            id="type-receiver-onesender"
+            v-model="$parent.notification.onesenderReceiver"
+            type="text"
+            placeholder="628123456789 or 628123456789-34534"
+            class="form-control"
+            required
+        />
+    </div>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:49]: is there a notification link avaliable to attach at the bottom like here or is the OneSender UI user friendly enough where these values can be found?

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Looks mostly good.

Could you please attach the screenshots we request in the contribution guide?
I have attached a template to the PR-Description.
The reason why we require this is that verififying that these notification providers is quite time-consuming for us and somewhat simple for you (as you already tested it).

File: src/components/notifications/Onesender.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        <label for="webhook-request-body" class="form-label">{{ $t("TypeOnesender") }}</label>
+        <select
+            id="webhook-request-body"
+            v-model="$parent.notification.onesenderTypeReceiver"
+            class="form-select"
+            required
+        >
+            <option value="private">{{ $t("PrivateOnesender") }}</option>
+            <option value="group">{{ $t("GroupOnesender") }}</option>
+        </select>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:36]: Could you please have a look if adding a helptext is benefitial for this option?
Selecting `private`/`group` in a selection field called `Group ID / Private` does not seem very obvious to endusers.
Feel free to overrule if it is better this way

File: src/components/notifications/Onesender.vue (hunk 1 above)
Comment (hasanbasri1993) [reply src/components/notifications/Onesender.vue:17]: i use password type input

File: src/components/notifications/Onesender.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
+<template>
+    <div class="mb-3">
+        <label for="host-onesender" class="form-label">{{ $t("HostOnesender") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:3]:
```suggestion
        <label for="host-onesender" class="form-label">{{ $t("Host Onesender") }}</label>
```

File: src/components/notifications/Onesender.vue [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            type="url"
+            placeholder="https://xxxxxxxxxxx.com/api/v1/messages"
+            pattern="https?://.+"
+            class="form-control"
+            required
+        />
+    </div>
+
+    <div class="mb-3">
+        <label for="receiver-onesender" class="form-label">{{ $t("TokenOnesender") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:16]:
```suggestion
        <label for="receiver-onesender" class="form-label">{{ $t("Token Onesender") }}</label>
```

File: src/components/notifications/Onesender.vue [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    <div class="mb-3">
+        <label for="webhook-request-body" class="form-label">{{ $t("TypeOnesender") }}</label>
+        <select
+            id="webhook-request-body"
+            v-model="$parent.notification.onesenderTypeReceiver"
+            class="form-select"
+            required
+        >
+            <option value="private">{{ $t("PrivateOnesender") }}</option>
+            <option value="group">{{ $t("GroupOnesender") }}</option>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:32]:
```suggestion
            <option value="private">{{ $t("Private Number") }}</option>
            <option value="group">{{ $t("Group ID") }}</option>
```

File: src/components/notifications/Onesender.vue [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    <div class="mb-3">
+        <label for="receiver-onesender" class="form-label">{{ $t("TokenOnesender") }}</label>
+        <HiddenInput id="receiver-onesender" v-model="$parent.notification.onesenderToken" :required="true" autocomplete="false"></HiddenInput>
+        <i18n-t tag="div" keypath="wayToGetOnesenderUrlandToken" class="form-text">
+            <a href="https://onesender.net/" target="_blank">{{ $t("here") }}</a>
+        </i18n-t>
+    </div>
+
+    <div class="mb-3">
+        <label for="webhook-request-body" class="form-label">{{ $t("TypeOnesender") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/Onesender.vue:24]:
```suggestion
        <label for="webhook-request-body" class="form-label">{{ $t("Recipient Type") }}</label>
```

File: src/lang/en.json [hunk 8]
```diff
@@ -962,5 +962,14 @@
...
+    "apiKeysDisabledMsg": "API keys are disabled because authentication is disabled.",
+    "HostOnesender": "Host Onesender",
+    "TokenOnesender": "Token Onesender",
+    "ReceiverOnesender": "Group ID / Number",
+    "TypeOnesender": "Recipient Type",
+    "PrivateOnesender": "Private Number",
+    "privateOnesenderDesc": "Make sure the number phone is valid. To send message into private number phone, ex: 628123456789",
+    "groupOnesenderDesc": "Make sure the GroupID is valid. To send message into Group, ex: 628123456789-342345",
+    "GroupOnesender": "Group ID",
+    "wayToGetOnesenderUrlandToken":"You can get the URL and Token by going to the Onesender website. More info {0}"
```
Comment (CommanderStorm) [inline src/lang/en.json:974]:
```suggestion
    "Host Onesender": "Host Onesender",
    "Token Onesender": "Token Onesender",
    "Recipient Type": "Recipient Type",
    "Private Number": "Private Number",
    "privateOnesenderDesc": "Make sure the number phone is valid. To send message into private number phone, ex: 628123456789",
    "groupOnesenderDesc": "Make sure the GroupID is valid. To send message into Group, ex: 628123456789-342345",
    "Group ID": "Group ID",
    "wayToGetOnesenderUrlandToken":"You can get the URL and Token by going to the Onesender website. More info {0}"
```

**Review by CommanderStorm — APPROVED** [review]
(There are a few changes that I am going to do, but those are "editorial comments")

Thanks for the new notification provider! 🎉 



> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>

**Pull request merged**